Spotify UI features

luccasclezar opened the pull request:
I love this tweak so I decided to make it more awesome 😁

By the way, I changed the iOS SDK version to 13.5 because I had it downloaded already and I don't know if it was a good idea, if it really wasn't, let me know.

- Background color based on album art.
- UI Hierarchy changes on the labels (size and color).
- Title of the track can extend to 2 lines.
- Option to remove Spotify icon and have more space to display the song's title.
- Option to remove margins on the album art.
- Code optimizations (the `updateMedia` function was resetting all views every time).

All this is Spotify-only as I don't use the other players.

luccasclezar committed:
Spotify optimizations and touch to open app

luccasclezar committed:
Add comments.

luccasclezar committed:
Small fix

luccasclezar committed:
Change background color with album art

luccasclezar committed:
Hide nextUpView if there's no queued song

luccasclezar committed:
Remove album art padding and re-add waveformView

luccasclezar committed:
Change text color based on background

luccasclezar committed:
Add widget options and some UI changes

- Toggle to remove album art margins.
- Toggle to remove Spotify icon.
- Some enhancements on the labels.

biD3V commented:
Hey thanks for the pull request! Couple of things:
- Can you make the coloring based on album art optional?
- Return the labels back to their original sizing.
- Make the Spotify logo and text green again (when not coloring the background).
- Return the constraints on the collection view back to the originals. Currently, this messes with the design. Though I understand that what you did might look better on smaller devices/icon layouts, overall it's less appealing.
- For some reason it seems the label view's leading constraint is set to zero in the small size but I can't find where that's happening.

These are the only "problems" I have with this, and thanks for the code optimizations as well!

luccasclezar commented:
> Make the Spotify logo and text green again (when not coloring the background).

I don't know how I didn't think of this, my bad. 😄

---

> Return the labels back to their original sizing.

Are you sure? Without a size difference, everything looks equally important, which, I think, isn't the case. The first thing you want to look is the song's title, then the artist, then the next up.

But of course, it's your tweak, I'm only asking 🙂

---

> For some reason it seems the label view's leading constraint is set to zero in the small size but I can't find where that's happening.

I noticed it too, gonna try to fix it.

biD3V commented:
> Without a size difference, everything looks equally important

I understand that, but I'm also trying to stay close to the original concept. Normally the title is colored so it sort of stands out. I do plan on messing with the font weights and sizes so they create a visual hierarchy of importance while also meeting my personal design standards. For now I'd like them to stay how I set them.

luccasclezar commented:
Ok then, already working on the changes 😁

luccasclezar committed:
Some changes based on feedback

- Optional background coloring.
- Revert changes on the UILabels sizes.
- Revert changes on the CollectionView's constraints.
- White waveform icon when the background is dark.
- Optimizations and organizational changes.

luccasclezar commented:
Sorry for taking this long, but I think it's all there.

Only thing is, I don't know why but both the Spotify icon and the SVG Waveform are not changing color based on background. I'm trying to fix it.

luccasclezar committed:
Fix appLabel size
